src/Avalonia.Visuals/VisualExtensions.cs

```diff
@@ -20,10 +19,8 @@ public static class VisualExtensions
         /// <returns>The point in client coordinates.</returns>
         public static Point PointToClient(this IVisual visual, PixelPoint point)
         {
-            var (root, offset) = GetRootAndPosition(visual);
-            var screenOffset = PixelPoint.FromPoint((Point)offset, root.RenderScaling);
-            var screenPoint = new PixelPoint(point.X - screenOffset.X, point.Y - screenOffset.Y);
-            return root.PointToClient(screenPoint);
+            var rootPoint = visual.VisualRoot.PointToClient(point);
+            return visual.VisualRoot.TranslatePoint(rootPoint, visual).Value;
```

Comment:
Maybe throw a null reference exception?

Reply:
Again, we know they're in the same tree as we're using `visual.VisualRoot`.